### PR TITLE
feat: show actual record size when throw maximum request size error

### PR DESCRIPTION
**Pull request by fraidev**
Related: https://github.com/infinyon/fluvio/issues/4216

### CODE REVIEW

**Review by digikata — APPROVED** [review]
LGTM, nice improvement on the err message

File: crates/fluvio/src/producer/error.rs [hunk 1]
```diff
@@ -9,8 +9,8 @@ use crate::producer::PartitionId;
 #[derive(thiserror::Error, Debug, Clone)]
 #[non_exhaustive]
 pub enum ProducerError {
-    #[error("the given record is larger than the max_request_size ({0} bytes)")]
-    RecordTooLarge(usize),
+    #[error("record size ({0} bytes), exceeded maximum request size ({0} bytes)")]
```
Comment (sehz) [inline crates/fluvio/src/producer/error.rs:12]: since api is changed, need to bump major version

**Review by sehz — CHANGES_REQUESTED** [review]

File: crates/fluvio/src/producer/error.rs (hunk 1 above)
Comment (fraidev) [reply crates/fluvio/src/producer/error.rs:12]: sure! I bumped the fluvio major version

**Review by sehz — APPROVED** [review]

**Comment by sehz** [comment]
@fraidev ,  fluvio `0.24` was not released so there was no need to bump to `0.25`.  can you revert back version number

**Pull request merged**